Content.Server/PDA/Ringer/RingerSystem.cs

```diff
@@ -0,0 +1,134 @@
+            ringer.isPlaying = true; //Allows the overridden Update function to play the ringtone based off of the ringer component's stored ringtonePathArr
+            UpdateRingerUserInterface(ringer);
+        }
+
+        private void UpdateRingerUserInterfaceDriver(EntityUid uid, RingerComponent ringer, RingerRequestUpdateInterfaceMessage args) { //Seprate Drivers are needed because UpdateRingerUserInterface() and UpdateRingerComponent() are called by other functions aside from the events themselves
+            UpdateRingerUserInterface(ringer);
+        }
+
+        private void RingerUpdateRingtoneDriver(EntityUid uid, RingerComponent ringer, RingerSetRingtoneMessage args) {
+            UpdateRingerComponent(ringer, args.ringtoneStringArgArr);
+        }
+
+        public void RandomizeRingtone(EntityUid uid, RingerComponent ringer, ComponentInit args) { //Randomizes the ringtone
+            string[] possibleNotes = new string[12]{"A","B","C","D","E","F","G","A#","C#","D#","F#","G#"};
+            string[] randomizedRingerStringArr = new string[4];
+            var randomNote = new Random();
+
+            for(int count = 0; count < 4; count++) {
+                int randomNoteIndex = randomNote.Next(possibleNotes.Length);
+                randomizedRingerStringArr[count] = possibleNotes[randomNoteIndex];
+            }
+
+            UpdateRingerComponent(ringer, randomizedRingerStringArr);
+
+        }
+
+        //Non Event Functions
+
+        private bool UpdateRingerComponent(RingerComponent ringer, string[] ringerStringArr) {
+            bool _isNull = false;
+            string[] _ringtonePathArr = new string[4];
+            for(int strngCount = 0; strngCount < 4; strngCount++) { //Prevents "null" (Not technically null but "" values cant be passed) values from being passed to the component
+                if(ringerStringArr[strngCount].Equals("")) {
+                    _isNull = true;
+                }
+            }
+            if(_isNull){return false;};
+            ringer.ringtoneStringArr = ringerStringArr;
+
+            for(int strngCount = 0; strngCount < 4; strngCount++) {
+                _ringtonePathArr[strngCount] = _pathIndex[ringerStringArr[strngCount]];
+            }
+
+            ringer.setRingerAudioPaths = _ringtonePathArr;
+            UpdateRingerUserInterface(ringer);
+
+            return true;
+        }
+
+        private void UpdateRingerUserInterface(RingerComponent ringer)
+        {
+            var ui = ringer.Owner.GetUIOrNull(RingerUiKey.Key);
+            ui?.SetState(new RingerUpdateState(ringer.isPlaying, ringer.ringtoneStringArr, ringer.setRingerAudioPaths));
+        }
+
+        public bool OpenRingerUI(RingerComponent ringer, IPlayerSession session)
+        {
+            var ui = ringer.Owner.GetUIOrNull(RingerUiKey.Key);
+            ui?.Toggle(session);
+            return true;
+        }
```

Comment:
Does the UI need to be disposed of?

Reply:
Yes, because if the PDA item is moved out of reach of the player, the player shouldn't have access to the menu anymore.